saleor/graphql/product/mutations/product_variant/product_variant_create.py

```diff
@@ -294,9 +296,20 @@ def check_for_duplicates_in_stocks(cls, stocks_data):
                 }
             )
 
+    @classmethod
+    def set_track_inventory(cls, _info, instance, cleaned_input):
+        track_inventory_by_default = get_track_inventory_by_default()
```

Comment:
Instead of calling SiteSettings.objects.all().first(), you should use `site = get_site_promise(info.context).get()`
Then you have access to site settings by making `site.settings`.
`get_site_promise` uses data loader to fetch the site settings, it means that we will fetch siteSettings only once per request (no matter if you need siteSettings in mutation or in query response). 
This will also resolve the issue in bulk mutation. You're using `get_track_inventory_by_default()` in loop. It means that you're adding additional db queryset for each variant in bulk

Reply:
@korycins Thanks for the suggestion. I've implemented the changes so basically use of data loaders is to reduce db queries . It will only query per requests. Is it something similar to caching but valid for one request only?

Reply:
Yes, correct. It is a batching algorithm connected with an in-memory cache. It solves the problem of fetching the same from the database in different graphql resolvers for a single request.